gk/main.c

```diff
@@ -1062,12 +977,13 @@ gk_synchronize(struct gk_fib *fib, struct gk_instance *instance)
 }
 
 static void
-process_gk_cmd(struct gk_cmd_entry *entry,
-	struct gk_instance *instance, struct gk_config *gk_conf)
+process_gk_cmd(struct gk_cmd_entry *entry, struct gk_add_policy **policies,
+	int *num_policies, struct gk_instance *instance)
 {
 	switch (entry->op) {
 	case GK_ADD_POLICY_DECISION:
-		add_ggu_policy(&entry->u.ggu, instance, gk_conf);
+		if (instance->num_scan_del == 0)
+			policies[(*num_policies)++] = &entry->u.ggu;
```

Comment:
Drop the test `if (instance->num_scan_del == 0)` here because the flow entry associated with this policy may be in the flow table. Leave the work of avoiding to add new entries when the flow table is full to `add_ggu_policy_bulk()` since the latter will check if the flow entry is already on the table.